typings/index.d.ts

```diff
@@ -869,30 +870,47 @@ export interface TelegrafOptions {
   /**
    * Telegram options
    */
-  telegram?: TelegramOptions
+  telegram?: TelegramOptions;
 
   /**
    * Bot username
    */
-  username?: string
+  username?: string;
+
+  /**
+   * Context Type
+   */
+  contextType?: ContextMessageUpdate;
+
+  /**
+   * retryAfter
+   */
+  retryAfter?: number;
+
+  /**
+   * handlerTimeout
+   */
+  handlerTimeout?: number;
 }
 
 export const Composer: ComposerConstructor;
 
+export type Handler<T extends ContextMessageUpdate> = Middleware<T> | Composer<T>
```

Comment:
This doesn't include for example `Router`.

```suggestion
interface MiddlewareObj<T extends ContextMessageUpdate> {
  middleware(): Middleware<T>;
}

export type Handler<T extends ContextMessageUpdate> = Middleware<T> | MiddlewareObj<T>
```

I'd also rename `Middleware` to `MiddlewareFn` and `Handler` to `Middleware`.